calico_rkt/calico_policy.py

```diff
@@ -0,0 +1,149 @@
+# Copyright 2015 Metaswitch Networks
+#
+# Licensed under the Apache License, Version 2.0 (the 'License');
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an 'AS IS' BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import logging
+from constants import *
+from calico_rkt import configure_logging
+from pycalico.datastore import DatastoreClient
+from pycalico.datastore_datatypes import Rule, Rules
+
+KUBERNETES_DEFAULT_PROFILE = "default_profile"
+
+_log = logging.getLogger(__name__)
+configure_logging(_log)
+
+class BasePolicyDriver(object):
+    """
+    Abstract base class of a CNI Policy Driver.
+
+    The CNI Policy Driver is responsible for applying network policy to
+    networked containers and creating/applying Calico profiles to Calico
+    endpoints.
+    """
+    def __init__(self):
+
+        self._client = DatastoreClient()
+        """
+        DatastoreClient for access to the Calico datastore.
+        """
+
+        self.profile_name = None
```

Comment:
I think this would be better as a @property.

e.g
```
@property
def profile_name(self):
  raise NotImplementedError("Must specify a profile name")
```

Then we don't need to override __init__ in the subclasses, and allows the subclasses to be a bit more creative with how they generate a profile name.